core/src/main/java/org/eclipse/hono/adapter/AdapterApplication.java

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright (c) 2016 Red Hat
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Red Hat - initial creation
+ */
+
+package org.eclipse.hono.adapter;
+
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import org.eclipse.hono.config.HonoConfigProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Base class for any Hono protocol adapter
```

Comment:
Is there anything about the implementation that makes it specific to implementing *protocol adapters*? Maybe we should better call it something like `VertxBasedApplication` to indicate that it is a Spring Boot application that requires a factory for producing a 'Verticle'?